### DISCUSSION TITLE
Trouble with appending hvplots in loop

### DISCUSSION THREAD

**Discussion by jsmi374** [earthlab-education/Earth-Analytics-2023-01-Intro#484] [Q&A06 Vector Data]
Hello, 

I am having trouble with creating a figure for the SJER study area ... specifically getting the hvplot loop to work with my python dictionary. 

I end up only getting one variable/ road type to show up in my figure. See the screen shot provided. Any hints would be much appreciated. 

![Screenshot 2023-10-09 121626](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/76793232/2a8ae353-c3d6-46e9-bfe8-069b684530d7)

**Comment by jsmi374** [comment]
This is an issue I was having about 2 weeks ago, and after our discussion in class on how to best present questions on the discussion board .. I realized this was maybe an opportunity to clean up this post. I actually closed this post shortly after I posted it because I worked out the answer with ChatGPT, but I never provided the potentially helpful information for others. here is my code that allowed me to plot with hvplots in a loop for this question: 
``` python 
# Define a colormap for your road types
roads_symb = {'M': 'black',  'Unknown': 'gray','C':'purple','S': 'green'}

# Create a DataFrame with your data
roads_df = roads.copy()

# Loop through each group (unique attribute value) in the roads layer and assign it a color
hv_objs = []
for ctype, data in roads_df.groupby('RTTYP'):
    color = roads_symb.get(ctype, 'gray')  # Use gray as default color
    hv_objs.append(data.hvplot.line(
        x='x', y='y', color=color, line_width=1, label=ctype))

# Combine the individual plots into a single plot
roads_plot = hv.Overlay(hv_objs).opts(
    title='Madera County Roads',aspect = 'equal',xaxis=None, yaxis=None)

boundary_plot = sjer_bound.hvplot(polygons='geometry', line_color='red', fill_color=None,line_width=2,label='SJER Boundary')

roads_plot*boundary_plot
```
![Screenshot 2023-10-20 194836](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/76793232/cd393912-7f83-40c0-b5cb-018039313ad9)

**Reply by eculler** [reply]
Thanks James - both the map and the revised post looks great!
